### PR TITLE
8305971: NPE in JavacProcessingEnvironment for missing enum constructor body

### PR DESCRIPTION
**Pull request by vicente-romero-oracle**
Please review this simple fix that is averting a NPE in JavacProcessingEnvironment if a an annotation processor is used while compiling this erroneous code:

```
enum T {
  ONE("");

  T(String one);  // missing constructor body

  final String one;
}
```
the fix is a one liner
TIA

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305971](https://bugs.openjdk.org/browse/JDK-8305971): NPE in JavacProcessingEnvironment for missing enum constructor body (**Bug** - P3)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to [222287d6](https://git.openjdk.org/jdk/pull/15974/files/222287d6f6b360cb7e289b84a47fb623e211ffd8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15974/head:pull/15974` \
`$ git checkout pull/15974`

Update a local copy of the PR: \
`$ git checkout pull/15974` \
`$ git pull https://git.openjdk.org/jdk.git pull/15974/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15974`

View PR using the GUI difftool: \
`$ git pr show -t 15974`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15974.diff">https://git.openjdk.org/jdk/pull/15974.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15974#issuecomment-1740032168)

### CODE REVIEW

**Comment by jddarcy** [comment]
Is this issue or an analagous issue triggered for default constructors or the special constructors of records?

Is so, it would be good to have explicit test coverage of those cases too.

File: test/langtools/tools/javac/annotations/crash_empty_enum_const/SimpleProcessor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.TypeElement;
+
+@SupportedAnnotationTypes("*")
```
Comment (jddarcy) [inline test/langtools/tools/javac/annotations/crash_empty_enum_const/SimpleProcessor.java:31]: The style of annotation processor tests does vary a bit by area of javac. It should be technically possible to put the code being tested and the annotation processor into a single file, as is commonly done in the direct annotation processing tests.

I can send some pointers if that kind of refactoring would be of interest.

**Comment by vicente-romero-oracle** [comment]
> Is this issue or an analagous issue triggered for default constructors or the special constructors of records?
> 
> Is so, it would be good to have explicit test coverage of those cases too.

I didn't see this issue being triggered for default constructors or record's special constructors.

File: test/langtools/tools/javac/annotations/crash_empty_enum_const/SimpleProcessor.java (hunk 1 above)
Comment (vicente-romero-oracle) [reply test/langtools/tools/javac/annotations/crash_empty_enum_const/SimpleProcessor.java]: > The style of annotation processor tests does vary a bit by area of javac. It should be technically possible to put the code being tested and the annotation processor into a single file, as is commonly done in the direct annotation processing tests.
> 
> I can send some pointers if that kind of refactoring would be of interest.

I have uploaded another iteration that refactors the test as suggested

**Review by jddarcy — APPROVED** [review]

**Pull request closed without merging**